### ISSUE TITLE
Show icons also in links

### ISSUE THREAD

**Issue by jckoester** [gfxholo/iconic#22]
When using iconize the icons get also displayed in front of the file link in any dataview table. Icons set by iconic don't. Is there a way to also show them?

**Comment by jckoester** [comment]
Actually this seems not to be related to dataview, this applies to all links. Icons defined by iconize get displayed in any link, icons from iconic not.

**Comment by jckoester** [comment]
Didn't read documentation, behavior is to be expected like it is.

**Comment by gfxholo** [comment]
Hi - it's true this feature isn't on the roadmap, but since I'm expecting a lot of "add this Iconize feature" requests over time, it'd be nice to keep this issue open so anyone can pitch in their comments.

You don't have to reply to this, but thanks for the suggestion! :)

**Comment by dog-in-space** [comment]
@gfxholo Would you be open to a pull request for this? And do you have a preference on any part of the implementation?

**Comment by gfxholo** [comment]
@dog-in-space Yeah, I'd definitely be open to a PR!

The way I'd go about it is by modifying [`EditorIconManager`](https://github.com/gfxholo/iconic/blob/9260f2810bae8252a9282e65a189296ea14375dd/src/EditorIconManager.ts), specifically using the [`constructor()`](https://github.com/gfxholo/iconic/blob/9260f2810bae8252a9282e65a189296ea14375dd/src/EditorIconManager.ts#L10) to set up any callbacks, and [`refreshIcons()`](https://github.com/gfxholo/iconic/blob/9260f2810bae8252a9282e65a189296ea14375dd/src/EditorIconManager.ts#L97) to apply DOM modifications (so this method can be called externally if the user changes an icon). You can fetch the current icon for a link by calling [`plugin.getFileItem(filepath)`](https://github.com/gfxholo/iconic/blob/main/src/IconicPlugin.ts#L519) which returns an object containing `icon` and `color`.

- For links in reading mode, you'd want to register a [markdown post-processor](https://docs.obsidian.md/Plugins/Editor/Markdown+post+processing).
- For editing mode, an [editor extension](https://docs.obsidian.md/Plugins/Editor/Editor+extensions) would be *best*, but I wouldn't mind plain MutationObservers either.

Just so you know, I'm wrapping up a pretty big update to `EditorIconManager` this week, so I'd look at the new patch when it drops and base any forks off that 🙏

**Comment by ShaneNZ** [comment]
Just weighing in that I would love to see this feature come to Iconic - the sole reason I'm still using Iconize is to have icons inline with the links in the text.

**Comment by gfxholo** [comment]
Yeah, you're not alone! Looks like this issue has nearly double the reactions of any other feature request. It's not a gigantic feature, so I'm waiting for a good moment to add this in a `1.1.x` patch 🪄🔮

**Comment by Kuuh4** [comment]
I feel like this FR is a huge improvement to the UX of obsidian!

**Comment by MysticMikester** [comment]
Upvote!! This!!

**Comment by henryfaber** [comment]
I recently discovered Iconize and I'm as impressed with your thoughtful UX implementation as I am with the actual functionality of the plugin. Well done!

I'm also very excited about the prospect of display the icon next to a link within a note's body. Would be super helpful!

**Comment by itsduncann** [comment]
Upvoting this feature as well! It's truly the only thing I miss from Iconize, it would help _so_ much!

**Comment by lbineau** [comment]
I'm also adding a comment to show my interest in the topic. @GRA0007 Why are you downvoting every single comment? Is it because there is another way to show interest that we are doing?

**Comment by GRA0007** [comment]
@lbineau As with any GitHub issue, the way to show your support is by reacting to the original post. That lets members sort issues by the amount of reactions they get to prioritise them.

People subscribe to updates so they can find out when there's _progress_, not when someone posts a "+1". By commenting on an issue you are notifying everyone who wants to know when there is discussion or an update about the issue. Please keep that in mind as it's common etiquette when interacting on this site.

**Comment by cmykenna** [comment]
I don't see this feature on the roadmap anymore. Apologies if I just overlooked it, but if not, is this feature still planned?

Would love to know the status of this as it's one of the major things keeping me using Iconize. 

I use icons heavily in my notes to quickly visually ID things, including note links, so having this feature is invaluable to me! (And to many others it seems, judging by the upvotes!)

Totally get you work on this in your free time and bug fixes probably take priority, but just wanted to ask as I couldn't find any recent info on when this might be added/if it's still planned.
